ntoskrnl/ke/amd64/cpu.c

```diff
@@ -35,37 +35,54 @@ static const CHAR CmpIntelID[]       = "GenuineIntel";
 static const CHAR CmpAmdID[]         = "AuthenticAMD";
 static const CHAR CmpCentaurID[]     = "CentaurHauls";
 
+#define CPUMODEL(eax) ((eax & 0xF0) >> 4) | ((eax & 0x000F0000) >> 12)
+#define CPUSTEPPING(eax) (eax & 0xF)
+#define CPUTYPE(eax) (eax & 0xF00) >> 8
+#define CPUTYPE_EXT(eax) ((eax & 0x0FF00000) >> 24) + 15
+
 /* FUNCTIONS *****************************************************************/
 
 VOID
 NTAPI
 KiSetProcessorType(VOID)
 {
     CPU_INFO CpuInfo;
-    ULONG Stepping, Type;
+
+    struct
+    {
+        union
+        {
+            USHORT SteppingAsUSHORT;
+            struct
+            {
+                UCHAR Stepping;
+                UCHAR Model;
+            };
+        };
+        UCHAR Family;
+    } CPU_REVISION;
```

Comment:
This structure is not a bitfield structure :sweat_smile: You can make it bitfield for the CPUID EAX register, and then you will not need to use the macros.

Reply:
Ah, I think I got it now. :)
Updated the PR.